ngraph/core/src/node.cpp

```diff
@@ -770,26 +770,32 @@ bool ov::Node::constant_fold(OutputVector& output_values, const OutputVector& in
     if (!all_constants)
         return false;
 
-    HostTensorVector input_tensors;
+    ov::runtime::TensorVector input_tensors;
     for (const auto& input : input_values) {
-        auto host_tensor = make_shared<ngraph::runtime::HostTensor>(
-            ov::as_type_ptr<ngraph::op::v0::Constant>(input.get_node_shared_ptr()));
-        input_tensors.push_back(host_tensor);
+        auto constant_node = ov::as_type<ngraph::op::v0::Constant>(input.get_node());
+        if (constant_node) {
+            auto host_tensor = ov::runtime::Tensor(constant_node->get_element_type(),
+                                                   constant_node->get_shape(),
+                                                   const_cast<void*>(constant_node->get_data_ptr()));
+            input_tensors.emplace_back(host_tensor);
+        }
     }
-    HostTensorVector output_tensors;
-    OutputVector output_constants;
+
+    ov::runtime::TensorVector output_tensors;
     for (const auto& output : outputs()) {
-        auto tensor = make_shared<HostTensor>(output.get_element_type(), output.get_partial_shape());
-        output_tensors.push_back(tensor);
+        auto host_tensor = ov::runtime::Tensor(output.get_element_type(), output.get_shape());
```

Comment:
It is not correct, because output shape can be dynamic.
Please add a code for dynamic shapes.
Example:
https://github.com/openvinotoolkit/openvino/blob/a115f2f4ea5cdee224e5fb0fac2f2a459fa61455/ngraph/core/src/function.cpp#L436

Reply:
added in commit 94306e1